### PR TITLE
Refine configurations for the vue-i18n eslint plugin

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Related to #1160 by @sarayourfriend

## Description

This PR:
- makes the i18n plugin ignore raw texts in test files, we don't need to check that rule there so this will get us rid of the recurring warnings
- makes this rule check to error 🔴 instead of warning
- deletes some comments for eslint exceptions as we can centralize specific texts to ignore from the eslint config file 🔥

It builds on top of #1239 that fixes the `lint` and `lint:fix` commands.

## Testing Instructions

Look at the **Files changed** tab of this PR and observe there are no warnings annotations. Try adding raw text different to brand names in templates, then run `pnpm lint` and observe the error is announced.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -161,6 +161,20 @@ module.exports = {
...
+      'error',
+      {
+        ignoreText: [
+          // Brand names that should not be translated
+          'Common Crawl',
+          'Creative Commons',
+          'Europeana',
+          'Flickr',
+          'Openverse',
+          'WordPress',
```
Comment (sarayourfriend) [inline .eslintrc.js:174]: Nice! This is a great find.

**Review by sarayourfriend — APPROVED** [review]

**Comment by sarayourfriend** [comment]
@krysal I just found out about the `--max-warnings` option for eslint, we can pass it as `--max-warnings=0` in the `lint` script and in the `lint-staged` script to prevent any warnings from sneaking into committed code without having to update the eslint configuration to change all the warnings to errors.

How do you feel about adding that to this PR (or to the base branch for this PR)? That way we can close #1160.

**Review by obulat — APPROVED** [review]
I love how cleaner the right side of the diff looks, @krysal ! Thank you for fixing this.

**Comment by krysal** [comment]
Great find @sarayourfriend! Added in [133c471](https://github.com/WordPress/openverse-frontend/pull/1252/commits/133c471706dac95c7d60d0b12b50df2b980e0f07).

**Pull request merged**